sdk/cosmosdb/cosmos/src/diagnostics/Diagnostics.ts

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+const _startTime = new Date().getTime();
+export const defaultHeaders: DiagnosticHeaders = {
+  cosmosdiagnostics: "Started Cosmos Diagnostics",
+  diagnosticStartTime: new Date().toLocaleString(),
+  durationInMs: new Date().getTime() - _startTime,
+};
+const _diagnosticHeaders: DiagnosticHeaders[] = [defaultHeaders];
```

Comment:
Shouldn't the assignment of default headers happen lazily vs. here? Otherwise the duration calculation would be incorrect? So, should be empty array or undefined here - then below you check the length and apply default headers?

Reply:
Done